File: Java/Funcionario.java
abstract class Funcionario {

    private String nome;
    private double salario;

    public Funcionario(String nome,double salario){
        this.nome = nome;
        this.salario = salario;
    }

    public abstract void calcularSalario();

    public String getNome(){
        return nome;
    }

    public double getSalario(){
        return salario;
    }
}

File: Python/Funcionario.py
from abc import ABC, abstractmethod

from FuncionarioAssalariado import FuncionarioAssalariado
from FuncionarioHorista import FuncionarioHorista


# Classe abstrata Funcionario
class Funcionario(ABC):
    def __init__(self, nome, salario):
        self.nome = nome
        self.salario = salario

    @abstractmethod
    def calcular_salario(self):
        pass

    def get_nome(self):
        return self.nome

    def get_salario(self):
        return self.salario

horista = FuncionarioHorista("João", 50, 160)
horista.calcular_salario()  # Saída: Salário do funcionário João é R$: 8000

assalariado = FuncionarioAssalariado("Maria", 5000)
assalariado.calcular_salario() 